webapp/stores/post_store.jsx

```diff
@@ -259,22 +260,32 @@ class PostStoreClass extends EventEmitter {
         this.postsInfo[id].postList = combinedPosts;
     }
 
+    // updates/adds all the instance of "post" in every postsInfo
     storePost(post, isNewPost = false) {
```

Comment:
@Zhouzi it looks like I'm having trouble getting any posts through. Wondering if it's related to the changes you described above? I worry there will be performance issues for large deployments, if `storePost` goes through all the `postsInfo` that contains it.

![image](https://cloud.githubusercontent.com/assets/13119842/23324328/0d501d78-fabb-11e6-9d00-89c690ad1d1a.png)

If it's easier, I'd propose instead removing the `Pin to channel` menu item from [...] when viewing a message in permalink view. This is because
 - it will involve fewer changes (and none to database)
 - the use case for pinning messages in permalink view is probably low (if not, we'll get feedback and we can add this functionality later)

Sorry that I'm proposing an alternate change, I wasn't aware of how significant the change would be.